src/NHibernate/AdoNet/AsyncExtensions.cs

```diff
@@ -0,0 +1,43 @@
+#if NETFX || NETSTANDARD2_0 || NETCOREAPP2_0
+using System;
+using System.Data;
+using System.Data.Common;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace NHibernate
+{
+	internal static class AsyncExtensions
+	{
+		public static Task<DbTransaction> BeginTransactionAsync(this DbConnection connection, CancellationToken cancellationToken = default)
```

Comment:
@maca88 do you know why async generator does not pick up these extensions for `DbConnection`?

Reply:
Async extension methods need to be registered by using `asyncExtensionMethods` option in `AsyncGenrator.yaml`. [Here](https://github.com/maca88/nhibernate-core/commit/61ed5ddab25c3f9fdbdb1b2a4c7455f3a93da612#diff-60e3e40b5223a1c12eeb3d32a9be6ceb498de4f79d2d62008da4a6330e30bfdfR247) you can see what changes are needed in the configuration to generate `BeginTransactionAsync` calls.

Reply:
Ok, thanks, I forgot about that.